features/events/traps/src/test/java/org/opennms/netmgt/trapd/TrapDTOMapperTest.java

```diff
@@ -0,0 +1,108 @@
+import org.opennms.netmgt.snmp.TrapNotification;
+import org.opennms.netmgt.snmp.snmp4j.Snmp4JTrapNotifier;
+import org.snmp4j.PDU;
+import org.snmp4j.mp.SnmpConstants;
+import org.snmp4j.smi.Integer32;
+import org.snmp4j.smi.IpAddress;
+import org.snmp4j.smi.Null;
+import org.snmp4j.smi.OID;
+import org.snmp4j.smi.OctetString;
+import org.snmp4j.smi.TimeTicks;
+import org.snmp4j.smi.VariableBinding;
+
+public class TrapDTOMapperTest {
+
+	@Test
+	public void object2dtoTest() throws UnknownHostException {
+
+		PDU snmp4JV2cTrapPdu = new PDU();
+
+		OID oid = new OID(".1.3.6.1.2.1.1.3.0");
+		snmp4JV2cTrapPdu.add(new VariableBinding(SnmpConstants.sysUpTime, new TimeTicks(5000)));
+		snmp4JV2cTrapPdu.add(new VariableBinding(SnmpConstants.snmpTrapOID, new OID(oid)));
+		snmp4JV2cTrapPdu.add(new VariableBinding(SnmpConstants.snmpTrapAddress,new IpAddress("127.0.0.1")));
+
+		snmp4JV2cTrapPdu.add(new VariableBinding(new OID(oid), new OctetString("Trap Msg v2-1")));
+		snmp4JV2cTrapPdu.add(new VariableBinding(new OID(oid), new OctetString("Trap Msg v2-2")));
+
+		snmp4JV2cTrapPdu.add(new VariableBinding(new OID("1.3.6.1.2.1.1.5.0"),
+				new OctetString("Trap v1 msg-1")));
+		snmp4JV2cTrapPdu.add(new VariableBinding(new OID(".1.3.6.1.2.1.1.3"),
+				new OctetString("Trap v1 msg-2")));
+		snmp4JV2cTrapPdu.add(new VariableBinding(new OID(".1.3.6.1.6.3.1.1.4.1.1"), 
+				new OctetString("Trap v1 msg-3")));
+		snmp4JV2cTrapPdu.add(new VariableBinding(new OID(".1.3.6.1.4.1.733.6.3.18.1.5.0"),
+				new Integer32(1))); 
+		snmp4JV2cTrapPdu.add(new VariableBinding(new OID("1.3.6.1.2.1.1.5.0"),
+				new Null())); 
+		snmp4JV2cTrapPdu.add(new VariableBinding(new OID("1.3.6.1.2.1.1.5.1"),
+				new Null(128)));
+		snmp4JV2cTrapPdu.add(new VariableBinding(new OID("1.3.6.1.2.1.1.5.2"),
+				new Null(129)));
+		snmp4JV2cTrapPdu.add(new VariableBinding(new OID("1.3.6.1.2.1.1.5.3"),
+				new Null(130)));
+		snmp4JV2cTrapPdu.setType(PDU.NOTIFICATION);
+
+		TrapInformation snmp4JV2cTrap = new Snmp4JTrapNotifier.Snmp4JV2TrapInformation(
+				InetAddressUtils.ONE_TWENTY_SEVEN, "public",
+				snmp4JV2cTrapPdu, new BasicTrapProcessor()
+		);
+
+		OnmsDistPoller distPoller = new OnmsDistPoller();
+		distPoller.setId(DistPollerDao.DEFAULT_DIST_POLLER_ID);
+		distPoller.setLabel(DistPollerDao.DEFAULT_DIST_POLLER_ID);
+		distPoller.setLocation("localhost");
+		DistPollerDao distPollerDao = new DistPollerDaoMinion(distPoller);
+
+		TrapObjectToDTOProcessor mapper = new TrapObjectToDTOProcessor();
+		mapper.setDistPollerDao(distPollerDao);
+
+		TrapDTO trapDto = mapper.object2dto(snmp4JV2cTrap);
+		System.out.println("trapDto is : " + trapDto);
+		System.out.println("trapDto.getBody() is : " + trapDto.getBody());
+		System.out.println("trapDto.getCommunity() is : " + trapDto.getHeader(TrapDTO.COMMUNITY));
+
+		TrapNotification snmp4JV2cTrap1 = TrapDTOToObjectProcessor.dto2object(trapDto);
+
+		// TODO: Add assertions to make sure that conversion worked
```

Comment:
Can we either implement this, or remove it? This test doesn't currently test anything.